ddcommon/src/azure_app_services.rs

```diff
@@ -0,0 +1,495 @@
+            query.get_var(FUNCTIONS_WORKER_RUNTIME),
+            query.get_var(FUNCTIONS_EXTENSION_VERSION),
+        ) {
+            (Some(_), Some(_)) => AzureContext::AzureFunctions,
+            (Some(_), None) => AzureContext::AzureFunctions,
+            (None, Some(_)) => AzureContext::AzureFunctions,
+            (None, None) => AzureContext::AzureAppService,
+        }
+    }
+
+    fn extract_subscription_id(s: Option<String>) -> Option<String> {
+        s?.split('+')
+            .next()
+            .filter(|s| !s.trim().is_empty())
+            .map(|v| v.to_string())
+    }
+
+    /*
+     * Computation of the resource id follow the same way the .NET tracer is doing:
+     * https://github.com/DataDog/dd-trace-dotnet/blob/834a4b05b4ed91a819eb78761bf1ddb805969f65/tracer/src/Datadog.Trace/PlatformHelpers/AzureAppServices.cs#L215
+     */
+    fn build_resource_id(
+        subscription_id: Option<&String>,
+        site_name: Option<&String>,
+        resource_group: Option<&String>,
+    ) -> Option<String> {
+        match (subscription_id, site_name, resource_group) {
+            (Some(id_sub), Some(sitename), Some(res_grp)) => Some(
+                format!(
+                    "/subscriptions/{}/resourcegroups/{}/providers/microsoft.web/sites/{}",
+                    id_sub, res_grp, sitename
+                )
+                .to_lowercase(),
+            ),
+            _ => None,
+        }
+    }
+
+    pub fn new<T: QueryEnv>(query: T) -> Self {
+        let is_relevant = query
+            .get_var(SERVICE_CONTEXT)
+            .map(|s| s.to_bool())
+            .unwrap_or(false);
+        let subscription_id =
+            AzureMetadata::extract_subscription_id(query.get_var(WEBSITE_ONWER_NAME));
+        let site_name = query.get_var(WEBSITE_SITE_NAME);
+
+        let (site_kind, site_type) = match AzureMetadata::get_azure_context(&query) {
+            AzureContext::AzureFunctions => ("functionapp".to_owned(), "function".to_owned()),
+            _ => ("app".to_owned(), "app".to_owned()),
+        };
+
+        let resource_group = query.get_var(WEBSITE_RESOURCE_GROUP);
+        let resource_id = AzureMetadata::build_resource_id(
+            subscription_id.as_ref(),
+            site_name.as_ref(),
+            resource_group.as_ref(),
+        );
+        let extension_version = query.get_var(SITE_EXTENSION_VERSION);
+        let operating_system = query.get_var(WEBSITE_OS);
+        let instance_name = query.get_var(INSTANCE_NAME);
+        let instance_id = query.get_var(INSTANCE_ID);
+
+        AzureMetadata {
+            resource_id,
+            subscription_id,
+            site_name,
+            resource_group,
+            extension_version,
+            operating_system,
+            instance_name,
+            instance_id,
+            is_relevant,
+            site_kind,
+            site_type,
+        }
+    }
+
+    pub fn is_relevant(&self) -> bool {
+        self.is_relevant
+    }
```

Comment:
What is the function `is_relevant` relevant to?

Reply:
is the AAS metadata is relevant => if we can use it